mongodb/activity/queryDocument/activity.go

```diff
@@ -0,0 +1,167 @@
+package queryDocument
+
+import (
+	ctx "context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/project-flogo/core/activity"
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	mongocon "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var logquery = log.ChildLogger(log.RootLogger(), "mongodb-querydocument")
+
+const (
+	connectionProp = "mongoConnection"
+	ivOperation    = "operation"
+	outputProperty = "Output"
+	ivInput        = "input"
+	ovCount        = "count"
+)
+
+func init() {
+	err := activity.Register(&Activity{}, New)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+// New functioncommon
+func New(ctx activity.InitContext) (activity.Activity, error) {
+	settings := &Settings{}
+	err := metadata.MapToStruct(ctx.Settings(), settings, true)
+	if err != nil {
+		return nil, err
+	}
+	if settings.Connection != "" {
+
+		mcon, toConnerr := coerce.ToConnection(settings.Connection)
+		if toConnerr != nil {
+			return nil, toConnerr
+		}
+		client := mcon.(*mongocon.MongodbSharedConfigManager).GetClient()
+		act := &Activity{client: client, operation: ctx.Settings()["operation"].(string), collectionName: ctx.Settings()["collName"].(string), database: mcon.(*mongocon.MongodbSharedConfigManager).GetClientConfiguration().Database}
```

Comment:
You already have "operation" in settings.Operation. You don't need to do ctx.Settings()["operation"].(string) (Unnecessary function call) same with "collName".

Reply:
Right.. will remove them and use settings object instead.